src/vs/workbench/contrib/files/common/explorerModel.ts

```diff
@@ -358,8 +358,11 @@ export class ExplorerItem {
 					}
 				}
 
-				for (const [_, dirEntryItem] of dirChildren.values()) {
-					items.push(dirEntryItem);
+				for (const [dirEntryName, dirEntryItem] of dirChildren.values()) {
+					const nestedItems = nested.get(dirEntryName);
```

Comment:
the metadata we deal with above (nesterParent, nestedChildren, etc. is important for supporting edit operations. When i run this code and try to drag and drop items around I end up in a broken state.

Also folders should never be nested parents, this doesn't seem to implement that restriction.